### PR TITLE
Fix off by one error

### PR DESCRIPTION
**Pull request by dschwoerer**
Clearly we need more tests for the FCI boundary operator.

### CODE REVIEW

**Comment by dschwoerer** [comment]
Cherry-picked from #3330 - which has a test that failed without this fix, as no boundaries where applied.

File: include/bout/boundary_region_iter.hxx [hunk 1]
```diff
@@ -540,7 +540,7 @@ public:
   signed char _valid() const { return region->bndry_points[pos].valid; }
   Ind3D _ind() const { return region->bndry_points[pos].index; }
   signed char _boundary_width() const {
-    return region->localmesh->ystart - region->bndry_points[pos].abs_offset;
+    return region->localmesh->ystart - region->bndry_points[pos].abs_offset + 1;
```
[the remark this answers is not in the document]
Comment (ZedThree) [reply include/bout/boundary_region_iter.hxx:543]: The `signed char` as a return type seems unnecessary, but we should at least have `static_cast` here

[the remark this answers is not in the document]
Comment (dschwoerer) [reply include/bout/boundary_region_iter.hxx:543]: :+1: Fair point, we later return anyway an int.

**Review by ZedThree — APPROVED** [review]
